library/src/main/android/permissions/dispatcher/PermissionUtils.java

```diff
@@ -9,13 +10,30 @@
 import android.provider.Settings;
 import android.support.annotation.Nullable;
 import android.support.v4.app.ActivityCompat;
+import android.support.v4.util.SimpleArrayMap;
 
 import static android.os.Build.VERSION_CODES.FROYO;
 import static android.os.Build.VERSION_CODES.GINGERBREAD;
+import static android.os.Build.VERSION_CODES.ICE_CREAM_SANDWICH;
+import static android.os.Build.VERSION_CODES.KITKAT_WATCH;
+import static android.os.Build.VERSION_CODES.JELLY_BEAN;
 import static android.support.v4.content.PermissionChecker.checkSelfPermission;
 
 public final class PermissionUtils {
 
+    // Map of dangerous permissions introduced in later framework versions.
+    // Used to conditionally bypass permission-hold checks on older devices.
+    private static final SimpleArrayMap<String, Integer> MIN_SDK_PERMISSIONS;
+    static {
```

Comment:
Yes, I hand-picked these permissions using the Android docs for Manifest.permission](http://developer.android.com/reference/android/Manifest.permission.html), extracting all permissions that were listed as "dangerous" **and** added in a later API version.